mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMailboxInconsistenciesService.java

```diff
@@ -0,0 +1,363 @@
+
+        private Context(AtomicLong processedMailboxEntries, AtomicLong processedMailboxPathEntries, AtomicLong fixedInconsistencies, Collection<ConflictingEntry> conflictingEntries, AtomicLong errors) {
+            this.processedMailboxEntries = processedMailboxEntries;
+            this.processedMailboxPathEntries = processedMailboxPathEntries;
+            this.fixedInconsistencies = fixedInconsistencies;
+            this.conflictingEntries = new ConcurrentLinkedDeque<>(conflictingEntries);
+            this.errors = errors;
+        }
+
+        long getProcessedMailboxEntries() {
+            return processedMailboxEntries.get();
+        }
+
+        long getProcessedMailboxPathEntries() {
+            return processedMailboxPathEntries.get();
+        }
+
+        long getFixedInconsistencies() {
+            return fixedInconsistencies.get();
+        }
+
+        ImmutableList<ConflictingEntry> getConflictingEntries() {
+            return ImmutableList.copyOf(conflictingEntries);
+        }
+
+        long getErrors() {
+            return errors.get();
+        }
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o instanceof Context) {
+                Context context = (Context) o;
+
+                return Objects.equals(this.processedMailboxEntries.get(), context.processedMailboxEntries.get())
+                    && Objects.equals(this.processedMailboxPathEntries.get(), context.processedMailboxPathEntries.get())
+                    && Objects.equals(this.fixedInconsistencies.get(), context.fixedInconsistencies.get())
+                    && Objects.equals(getConflictingEntries(), getConflictingEntries())
+                    && Objects.equals(this.errors.get(), context.errors.get());
+            }
+            return false;
+        }
+
+        @Override
+        public final int hashCode() {
+            return Objects.hash(processedMailboxEntries.get(), processedMailboxPathEntries.get(), fixedInconsistencies.get(), getConflictingEntries(), errors.get());
+        }
+    }
+
+    private final CassandraMailboxDAO mailboxDAO;
+    private final CassandraMailboxPathV2DAO mailboxPathV2DAO;
+    private final CassandraSchemaVersionDAO versionDAO;
+
+    @Inject
+    SolveMailboxInconsistenciesService(CassandraMailboxDAO mailboxDAO, CassandraMailboxPathV2DAO mailboxPathV2DAO, CassandraSchemaVersionDAO versionDAO) {
+        this.mailboxDAO = mailboxDAO;
+        this.mailboxPathV2DAO = mailboxPathV2DAO;
+        this.versionDAO = versionDAO;
+    }
+
+    Mono<Result> fixMailboxInconsistencies(Context context) {
+        assertValidVersion();
+        return Flux.merge(
+                processMailboxDaoInconsistencies(context),
+                processMailboxPathDaoInconsistencies(context))
+            .reduce(Result.COMPLETED, Task::combine);
+    }
+
+    private void assertValidVersion() {
+        Optional<SchemaVersion> maybeVersion = versionDAO.getCurrentSchemaVersion().block();
```

Comment:
should throw on version not found?

Reply:
That's already the case as assertValidVersion throws on "version not found"